test/test_slimta_policy_headers.py

```diff
@@ -63,5 +65,53 @@ def test_add_received_header_prepended(self):
         AddReceivedHeader().apply(env)
         self.assertEqual(['Received', 'From'], env.headers.keys())
 
+    def test_add_dkim_header(self):
+        log = []
+        def set_log(a,b,c,d, **kwargs):
+            log.append(d)
+        self.mox.StubOutWithMock(slimta.logging, 'logline')
+        slimta.logging.logline(IgnoreArg(), IsA(str), IsA(int), IsA(str),
+                sender='sender').WithSideEffects(set_log)
+        slimta.logging.logline(IgnoreArg(), IsA(str), IsA(int),
+                IsA(str)).WithSideEffects(set_log)
+        slimta.logging.logline(IgnoreArg(), IsA(str), IsA(int),
+                IsA(str), domain=b'example.com',
+                sender='sender@example.com').WithSideEffects(set_log)
+        self.mox.ReplayAll()
+        env = Envelope('sender', ['rcpt@example.com'])
+        env.parse(b'From: test@example.com\r\n')
+        dkim = {}
+        AddDkimHeader(dkim).apply(env)
+        self.assertFalse(env.headers['DKIM-Signature'])
+        self.assertTrue(log[0] == "DKIM: invalid sender")
+        env = Envelope('sender@example.com', ['rcpt@example.com'])
+        env.parse(b'From: test@example.com\r\n')
+        AddDkimHeader(dkim).apply(env)
+        self.assertFalse(env.headers['DKIM-Signature'])
+        self.assertTrue(log[1] ==
+                "DKIM: domain :'example.com' is not setup, ignore")
+        dkim['example.com'] = { 'privkey': 'bad', 'selector': 'sel',
+                'signature_algorithm': 'rsa-sha1',
+                'include_headers': ['from','subject'] }
+        AddDkimHeader(dkim).apply(env)
+        self.assertFalse(env.headers['DKIM-Signature'])
+        self.assertTrue(log[2] == "DKIM: exception:Private key not found")
+        pk = """
+-----BEGIN RSA PRIVATE KEY-----
+MIIBOwIBAAJBAMhmwtECnKod9ywj8KcK308anyuS2iglAAoaAsibaduk0TTZX/sG
+wOAwwh71jsrdMIMDKGAnOn7ikYSVfxvFUQsCAwEAAQJAVWVsyRIa3mcsh9O83gHF
+DPlkMHZAnnC95pAU9ZU8c8qzGolDz2h3g+3py09L2dNN1KrmHgjs706OuKznTK3C
+KQIhAPocPlKGOQvM5t1Iv7kU2dkMsDso5iMLWJ7si5zTAM7/AiEAzR7aoUhJiFaD
+gm35ak2QzAk99H6uZXL5pPCvQJ+HyfUCIQCxTXJU2Df6iJAk0JyxXPmuJ5OK7Mxw
+jWuOhgvW6bIKCwIhAJ7RT+hmpwCYM7TuX5puOjmwwjanS3KjRiXucVMw8httAiBV
+LB94QlyDoRo6NOTbRIU1quGV/G3jufSl5hgqwuibQw==
+-----END RSA PRIVATE KEY-----
+"""
+        dkim['example.com'] = { 'privkey': pk, 'selector': 'sel',
+                'signature_algorithm': 'rsa-sha1',
+                'include_headers': ['from','subject'] }
+        AddDkimHeader(dkim).apply(env)
+        self.assertTrue(env.headers['DKIM-Signature'])
```

Comment:
Can you add a few `assertIn` checks for some of the expected tag=value pairs in the header?